presentation/src/main/java/me/androidbox/presentation/event/screen/EventScreen.kt

```diff
@@ -87,110 +81,219 @@ fun EventScreen(
+                            )
+                            Spacer(modifier = Modifier.height(16.dp))
                         }
-                    )
 
-                    Spacer(modifier = modifier.height(26.dp))
+                        if(eventScreenState.filteredVisitorsNotGoing.isNotEmpty()) {
+                            items(eventScreenState.filteredVisitorsNotGoing) { attendee ->
+                                VisitorItem(
+                                    initials = attendee.fullName.toInitials(),
+                                    userId = attendee.userId,
+                                    fullName = attendee.fullName,
+                                    isCreator = eventScreenState.host == attendee.userId,
+                                    onDeleteClicked = { userId ->
+                                        eventScreenEvent(EventScreenEvent.OnDeleteVisitor(userId))
+                                    }
+                                )
+                            }
+                        }
+                    }
 
-                    AgendaAction(
-                        agendaActionType = if(eventScreenState.isUserEventCreator) { AgendaActionType.DELETE_EVENT } else { AgendaActionType.JOIN_EVENT },
-                        onActionClicked = { agendaActionType ->
-                            when(agendaActionType) {
-                                AgendaActionType.DELETE_EVENT -> {
-                                    eventScreenEvent(EventScreenEvent.OnShowDeleteEventAlertDialog(shouldShowDeleteAlertDialog = true))
-                                }
-                                AgendaActionType.JOIN_EVENT -> {
+                    item {
+                        Spacer(modifier = modifier.height(26.dp))
+                        AgendaAction(
+                            agendaActionType = if (eventScreenState.isUserEventCreator) {
+                                AgendaActionType.DELETE_EVENT
+                            } else {
+                                /** Based on if the attendee has updated isGoing status
+                                 * isGoing = true ==> LEAVE_EVENT
+                                 * isGoing = false ==> JOIN_EVENT
+                                 * */
 
-                                }
-                                AgendaActionType.LEAVE_EVENT -> {
+                                /* Find the attendee and check their status - TODO Consider a helper method for this */
+                                val isAttendeeGoing = eventScreenState.attendees.firstOrNull { attendee ->
+                                    attendee.userId == eventScreenState.currentLoggedInUserId
+                                }?.isGoing ?: false
+
+                                if(isAttendeeGoing)
+                                    AgendaActionType.LEAVE_EVENT
+                                else
+                                    AgendaActionType.JOIN_EVENT
```

Comment:
I would either add the {}, or make it all in one line

It's a bit easier to read (+ it's a best practice on Android's guidelines https://developer.android.com/kotlin/style-guide#formatting )

Reply:
@MKiperszmid Updated